flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/AggregateFunction.scala

```diff
@@ -17,36 +17,93 @@
  */
 package org.apache.flink.table.functions
 
-import java.util.{List => JList}
-
-import org.apache.flink.api.common.typeinfo.TypeInformation
-import org.apache.flink.table.api.TableException
-
 /**
   * Base class for User-Defined Aggregates.
   *
-  * @tparam T the type of the aggregation result
+  * The behavior of an [[AggregateFunction]] can be defined by implementing a series of custom
+  * methods. An [[AggregateFunction]] needs at least three methods: createAccumulator, getValue,
+  * and accumulate. There are a few other methods that can be optional to have: retract, merge,
+  * resetAccumulator, and getAccumulatorType.
+  *
+  * All these methods muse be declared publicly, not static and named exactly as the names
+  * mentioned above. The methods createAccumulator and getValue are defined in the
+  * [[AggregateFunction]] functions, while other methods are explained below.
+  *
+  *
+  * {{{
+  * Processes the input values and update the provided accumulator instance. The method
+  * accumulate can be overloaded with different custom types and arguments. This function is
```

Comment:
"This function is always a MUST to have." -> "An AggregateFunction requires at least one accumulate method"?